vyper/semantics/types/module.py

```diff
@@ -117,8 +117,9 @@ def _is_function_implemented(fn_name, fn_type):
 
             to_compare = fns_by_name[fn_name]
             assert isinstance(to_compare, ContractFunctionT)
+            assert isinstance(fn_type, ContractFunctionT)
 
-            return to_compare.implements(fn_type)
+            return fn_type.implements(to_compare)
```

Comment:
seems wrong! compare_type is not commutative

Reply:
reverted back